test/cpp/end2end/ssl_credentials_test.cc

```diff
@@ -162,153 +164,23 @@ class SslCredentialsTest : public ::testing::TestWithParam<SslOptions> {
     return context.auth_context();
   }
 
-  void ExpectResumedSession(std::shared_ptr<const AuthContext> auth_context) {
+  static absl::string_view GetAuthContextProperty(
+      const AuthContext& auth_context, const std::string& property) {
     std::vector<grpc::string_ref> properties =
-        auth_context->FindPropertyValues(GRPC_SSL_SESSION_REUSED_PROPERTY);
-    ASSERT_EQ(properties.size(), 1u);
-    EXPECT_EQ("true", ToString(properties[0]));
+        auth_context.FindPropertyValues(property);
+    if (properties.size() != 1) return "";
+    return absl::string_view(properties[0].data(), properties[0].size());
   }
 
-  void ExpectNonResumedSession(
-      std::shared_ptr<const AuthContext> auth_context) {
+  static absl::flat_hash_set<std::string> GetAuthContextPropertyAsSet(
```

Comment:
Thanks Mark and apologies for the long-delayed reply.

Agreed with your general approach, a few minor changes in execution.

- Changed to return a list.
- Changed to avoid duplicating functionality, except the implementation turned out to be simpler if `GetAuthContextPropertyAsList` has the logic to parse out the properties from the auth context and then we can call this function to implement `GetAuthContextProperty`.
- Done RE `ToString`.
- Done RE `absl::string_view`.
- Re using `absl::StrSplit`: no longer applies since I inverted the order of which function calls which.

Reply:
This approach looks good!  I think I was misreading the code earlier when I suggested `StrSplit()` -- we weren't actually splitting a string but rather returning a list when the property was present more than once.  Sorry for the confusion.